### ISSUE TITLE
Streaming Linux to Linux - Mouse far too sensitive (using laptop trackpad?) - Witcher 2, Goat Simulator, ...

### ISSUE THREAD

**Issue by apocalyptech** [ValveSoftware/steam-for-linux#3484]
I'm not sure if this would be a bug in Witcher 2 or in Steam, but when I use In-Home streaming (from Linux to Linux), the mouse in Witcher 2 becomes extraordinarily sensitive, to the point where it's quite difficult to use the menus.  The in-game mouse sensitivity setting was already quite low, and setting it to its minimum level doesn't really help much.

I've started up a few other games (XCOM, Portal 2, etc) and the mouse was fine in those, so it's something related to just that game, though I suppose it's possible it could be happening in others as well.

Both are on steam package version 1410299342.

**Comment by Chizbang** [comment]
Any updates on this? This issue is soooo annoying and itd be great to have someone on this.

**Comment by justin8** [comment]
Over a year later, and this bug is still open? Hello Valve...

**Comment by bdowning** [comment]
Yep, me too.  And only my trackpad, if I plug in my USB mouse it's okay.  Streaming Terraria (yes, I know it runs native on Linux now, but it's a good example) is a good example of a completely uncontrollable game with a trackpad.

A couple of theories I tried:

* Wondered if it had something to do with report rate (i.e. the trackpad is reporting faster than the mouse, and the other side is accelerating per event or something).  Doesn't seem to be the case, as my mouse reports much faster than my trackpad:

```
SynPS/2 Synaptics TouchPad: Latest   100Hz, Average   100Hz
SynPS/2 Synaptics TouchPad: Latest   166Hz, Average   101Hz
SynPS/2 Synaptics TouchPad: Latest    71Hz, Average   100Hz
SynPS/2 Synaptics TouchPad: Latest   100Hz, Average   100Hz
SynPS/2 Synaptics TouchPad: Latest    50Hz, Average   100Hz
SynPS/2 Synaptics TouchPad: Latest   100Hz, Average   100Hz
SynPS/2 Synaptics TouchPad: Latest   100Hz, Average   100Hz
```

```
Razer  Razer Orochi 2013: Latest  1000Hz, Average   909Hz
Razer  Razer Orochi 2013: Latest   500Hz, Average   901Hz
Razer  Razer Orochi 2013: Latest  1000Hz, Average   901Hz
Razer  Razer Orochi 2013: Latest  1000Hz, Average   921Hz
Razer  Razer Orochi 2013: Latest  1000Hz, Average   921Hz
Razer  Razer Orochi 2013: Latest  1000Hz, Average   921Hz
Razer  Razer Orochi 2013: Latest  1000Hz, Average   929Hz
```

* Wondered if it had to do with accel curves.  Usually trackpads have a device-specific acceleration curve enabled by default ("Device Accel Profile: 1" in xinput).  I tried setting it to the standard X curve of 0 (also used by my mouse) to no effect.

```
:; xinput set-prop 'SynPS/2 Synaptics TouchPad' 'Device Accel Profile' 0
```

* Further, xinput acceleration settings seem to have no effect whatsoever.  Tried setting the constant deceleration factor (default 2.5 on my trackpad) very high, so it is unusably slow (maybe 10 pixels per inch).  Still uncontrollably fast in streaming.

```
:; xinput set-prop 'SynPS/2 Synaptics TouchPad' 'Device Accel Constant Deceleration' 100
```

* Wondered if somehow it was reading the trackpad directly in addition to through X.  "Floated" the trackpad device, removing it from controlling the X pointer.  Doesn't work at all in steam streaming anymore:

```
:; xinput float 'SynPS/2 Synaptics TouchPad'
```

Looking at the raw evdev output isn't helpful, as the touchpad is reported in absolute coordinates and pressure data.  The X11 synaptics driver must convert this into relative motion information.

Anyway, I did some legwork.  I'd try to do more, but obviously I can't look inside the streaming client.  Please try to fix this or at least acknowledge what you think might be going on.  It's really irritating having to plug a mouse into my laptop to be able to stream anything that doesn't use a hardware cursor.  Thanks, @slouken , @gdrewb-valve , Valve.

**Comment by DoogeJ** [comment]
Same issue here with Linux (Debian Jessie + Native steam client on a HP EliteBook 8730w) to Windows 10 while playing Homeworld 2 Remastered.

People on the forums report a workaround with enabling hardware cursor (obviously won't work for mouse-controlled-POV games). Sadly HW2 doesn't have that option.

**Comment by apocalyptech** [comment]
Interesting!  So today I installed Borderlands 2 directly on my laptop -- I'd ordinarily just use in-home streaming for it, but I'm going to be away from home this weekend.  B2 was one of the games which had the Synaptics super-sensitive issue when in-home-streaming.  As it turns out, the synaptics touchpad was actually waaay too sensitive when running the game locally, as well!

I wonder if it's actually an issue with the way the driver talks to the games themselves, rather than something specific with in-home streaming.

**Comment by installgentoo** [comment]
the bug is still there

**Comment by Managor** [comment]
Is this still an issue with modern Linux and Steam?

**Comment by Managor** [comment]
@kisak-valve This can be assumed to be fixed.

**Comment by hxka** [comment]
This can absolutely NOT be assumed to be fixed.

When I connect from my laptop (3200x2000) to my desktop (2560x1440), in games that use software cursor it is unusably slow when using a touchpad. Hardware cursor works fine. When using a mouse, both hardware and software cursors behave identically and correctly.

I originally thought that monitor scaling might be related (I use 2x on the laptop and 1x on the desktop), but even when I disable scaling and adjust cursor speed to compensate, the software cursor is still slower while streaming.)

**Comment by Managor** [comment]
Thanks. Good to hear from someone that can still test this.

**Comment by hxka** [comment]
I should say I use KDE Wayland at both computers. Setting screen resolution on the client to 1600x1000 did not affect the issue, so it seems it is not related.

What is related is touchpad settings. After experimenting, it seems software cursor always behaves as if I set pointer speed to 0 and disabled pointer acceleration.
